src/lib/permutations/keccak_fips/keccak_fips.h

```diff
@@ -0,0 +1,126 @@
+/*
+* Keccak-FIPS
+* (C) 2010,2016 Jack Lloyd
+* (C) 2023 Falko Strenzke
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_KECCAK_FIPS_H_
+#define BOTAN_KECCAK_FIPS_H_
+
+#include <botan/secmem.h>
+#include <string>
+
+namespace Botan {
+
+/**
+* KECCAK FIPS
+*
+* This file implements Keccak[c] which is specified by NIST FIPS 202 [1], where
+* "c" is the variable capacity of this hash primitive. Keccak[c] is not  a
+* general purpose hash function, but used as the basic primitive for algorithms
+* such as SHA-3 and KMAC. This is not to be confused with the "informal" general purpose hash
+* function which is referred to as "Keccak" and apparently refers to the final
+* submission version of the Keccak submission in the SHA-3 contest, possibly
+* what is released by NIST under the name "KECCAK - Final Algorithm Package" [2].
+* See also the file keccak.h for the details how the keccak hash function is defined
+* in terms of the Keccak[c] – a detail which cannot be found in [1].
+*
+*
+*
+* [1] FIPS PUB 202 – FEDERAL INFORMATION PROCESSING STANDARDS PUBLICATION – SHA-3 Standard: Permutation-Based Hash and Extendable-Output Functions
+*       https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf#page=28
+* [2] https://csrc.nist.gov/projects/hash-functions/sha-3-project
+*/
+
+class Keccak_FIPS final {
+   public:
+      /**
+      * @param output_bits the size of the hash output; must be one of
+      *                    224, 256, 384, or 512
+      */
+      explicit Keccak_FIPS(size_t output_bits, size_t capacity_bits, uint64_t custom_padd, uint8_t custom_padd_bit_len);
+
+      size_t hash_block_size() const { return m_bitrate / 8; }
+
+      size_t output_length() const { return m_output_bits / 8; }
+
+      size_t output_bits() const { return m_output_bits; }
```

Comment:
Probably  these should be removed. See the other comments re: "output" and `expand()`.